[MISC] Add more information to the bit manipulation task. (#12)
seqan/learning-resources: doc/nanocourse/bitvectors.md

### Access the compact bitvector

> The bitvector needs to support access to the single bits by a member function `read(i)`,

Reviewer: ```suggestion
The bitvector needs to support access to the single bits via a member function `read(i)`,
```

which returns either `0` or `1` depending on the bit at position `i`.

> We now face the problem that we do not store single bits but groups of 64bits in one `64bit` integer.

Reviewer: ```suggestion
We now face the problem that we do not store single bits but groups of 64 bits in one `uint64_t` integer.
```

?

> For example, if the data vector would contain the entry `17`, the 64bit binary representation of `17` is

Reviewer: ```suggestion
For example, if the data vector would contain the entry `17`, the 64 bit binary representation of `17` is
```

Also elsewhere

```
0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0001 0001
```

> It thus represents a group of 64 bits where the bits at position 59 and 63 (starting from 0) are set to 1.

Reviewer: ```suggestion
It thus represents a group of 64 bits where the bits at position 59 and 63 (starting from the left and 0) are set to 1.
```

0
```

If you are unexperienced with C++ we provide the backbone of read function and you can fill in the *TODOs*:

\hint
```cpp
struct Bitvector
{
    std::vector<uint64_t> data;

    Bitvector(size_t const count)
    {

>         data.resize((count + 63) / 64); // the +63 are a trick to round up the fraction.

Reviewer: ```suggestion
        data.resize((count + 63) / 64); // the +63 is a trick to round up the fraction.
```

    }

    bool read(size_t const i) const
    {
        // We need to find the correct group for bit i.
        // For example, if i is 10 it will be the 10th bit in the first 64bit-group,
        // if i is 70 it will be the 6th bit in the second.
        size_t group_index = /*TODO: In which data entry (group) can we find bit i?*/;

>         // Now that we identified the correct entry. We nee to know the relative position of i within its group,

Reviewer: ```suggestion
        // Now that we identified the correct entry. We need to know the relative position of i within its group,
```

### Write to the compact bitvector

We now want to support writing to our bitvector by implementing the member function `write(i, x)`.

> This is a bit more tricky so we recommend this is a bonus question for experienced C++ users.

Reviewer: ```suggestion
This is a bit more tricky so we recommend this as a bonus question for experienced C++ users.
```
